Detections/Dynamics365Activity/NewOfficeUserAgentinDynamics.yaml

```diff
@@ -0,0 +1,36 @@
+id: 572f3951-5fa3-4e42-9640-fe194d859419
+name: New Office User Agent in Dynamics 365 
+description: |
+  'Detects users accessing Dynamics from a User Agent that has not been seen in any Office 365 workloads in the last 7 days. Has configurable filter for known good user agents such as PowerApps.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: Dynamics365
+    dataTypes:
+      - Dynamics365Activity
+queryFrequency: 1d
+queryPeriod: 7d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  let timeframe = 1h;
+  let lookback = 7d;
+  let known_useragents = dynamic([]);
+  Dynamics365Activity
+  | where TimeGenerated > ago(timeframe)
+  | extend Message = split(OriginalObjectId, ' ')[0]
```

Comment:
I think you need to cast tostring here.

Reply:
On what element here? OriginalObjectId is string type already (do you mean make Message string?